.github/workflows/update-js-deps.yml

```diff
@@ -0,0 +1,139 @@
+          echo "github.ref_name: ${{ github.ref_name }}"
+          echo "github.ref_type: ${{ github.ref_type }}"
+          echo "github.repository: ${{ github.repository }}"
+          echo "github.sha: ${{ github.sha }}"
+          echo "job.status: ${{ job.status }}"
+          # https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows
+          echo "github.event.issue.pull_request: ${{ github.event.issue.pull_request }}"
+          echo "github.event.issue.number: ${{ github.event.issue.number }}"
+          echo "github.event.repository.name: ${{ github.event.repository.name }}"
+
+  js-upgrade:
+    name: 'JS upgrade'
+    runs-on: ubuntu-latest
+    env:
+      pr_title: DEP Update JS dependencies
+      pr_desc: Automated yarn upgrade and yarn build
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Read .nvmrc
+        run: echo ::set-output name=version::$(cat .nvmrc)
+        id: read-nvm
+
+      - name: Setup node
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ steps.read-nvm.outputs.version }}
+
+      - name: Install yarn
+        run: npm install --global yarn
+
+      - name: Install admin JS
+        if: github.event.repository.name != 'silverstripe-admin'
+        run: |
+          # Install admin js in sibling directory so shared components are available
+          DIR=$(pwd)
+          cd ..
+          git clone https://github.com/silverstripe/silverstripe-admin.git
+          mv silverstripe-admin admin
+          cd admin
+          git checkout 1
+          yarn install
+          cd $DIR
+      - name: Yarn upgrade
+        run: yarn upgrade
+
+      - name: Read package.json
+        id: package-json
+        run: |
+          # Read package.json to see if lint and test are runnable scripts
+          LINT=0
+          TEST=0
+          if [ "$(jq .scripts.lint? package.json)" != "null" ]; then LINT=1; fi
+          if [ "$(jq .scripts.test? package.json)" != "null" ]; then TEST=1; fi
+          echo "::set-output name=lint::$LINT"
+          echo "::set-output name=test::$TEST"
+          echo "LINT is $LINT"
+          echo "TEST is $TEST"
+
+      # The following 3 steps make up `yarn build`
+      # Splitting apart to make it easier to see where any failures originate from
+      - name: Yarn lint
+        if: steps.package-json.outputs.lint == 1
+        run: yarn lint
+
+      - name: Yarn test
+        if: steps.package-json.outputs.test == 1
+        run: yarn test
+
+      - name: Build JS with webpack
+        # always run this and subsequent steps even if yarn.lint/yarn.test fails so that pull-request is
+        # created which will result in a red pull-request build so it's easy to see where things are at
+        if: always()
+        run: NODE_ENV=production node_modules/webpack/bin/webpack.js -p --bail --progress
+
+      - name: Git
+        if: always()
+        run: |
```

Comment:
There's a lot of [GitHub actions steps to create pull request](https://github.com/marketplace?type=actions&query=Create+Pull+request) out there.

My preference would be to reuse existing steps as much as possible rather than re-invent the wheel.

Reply:
"Re-invent" is a bit of a stretch since this code was copied from the [official docs](https://docs.github.com/en/rest/reference/pulls#create-a-pull-request)

I'd rather not use another github action before
a) Need to spend time researching and implementing, meanwhile this solution already works
b) Ensure that PRs will be created the user as the 'github-actions' user
c) We'd still need to retain the existing code for `# Remove any existing pull-requests via GitHub API`

Reply:
Have moved the 'create a pull request' to a new action on emteknetnz/gha-pull-request